url7f729337e3_timmolter_Sundial_tgz-pJ8-org_knowm_sundial_SampleRunJ8/src/org/knowm/sundial/ee/SundialInitializerListener.java

```diff
@@ -0,0 +1,167 @@
+/**
+ * Copyright 2015 Knowm Inc. (http://knowm.org) and contributors.
+ * Copyright 2013-2015 Xeiam LLC (http://xeiam.com) and contributors.
+ * Copyright 2001-2011 Terracotta Inc. (http://terracotta.org).
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.knowm.sundial.ee;
+
+import javax.servlet.ServletContext;
+import javax.servlet.ServletContextEvent;
+import javax.servlet.ServletContextListener;
+
+import org.knowm.sundial.SundialJobScheduler;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * <p>
+ * A ServletContextListner that can be used to initialize Sundial.
+ * </p>
+ * <p>
+ * The init parameter 'quartz:shutdown-on-unload' can be used to specify whether you want scheduler.shutdown() called when the listener is unloaded
+ * (usually when the application server is being shutdown). Possible values are "true" or "false". The default is "true".
+ * </p>
+ * <p>
+ * The init parameter 'quartz:wait-on-shutdown' has effect when 'quartz:shutdown-on-unload' is specified "true", and indicates whether you want
+ * scheduler.shutdown(true) called when the listener is unloaded (usually when the application server is being shutdown). Passing "true" to the
+ * shutdown() call causes the scheduler to wait for existing jobs to complete. Possible values are "true" or "false". The default is "false".
+ * </p>
+ * <p>
+ * The init parameter 'quartz:start-on-load' can be used to specify whether you want the scheduler.start() method called when the listener is first
+ * loaded. If set to false, your application will need to call the start() method before the scheduler begins to run and process jobs. Possible values
+ * are "true" or "false". The default is "true", which means the scheduler is started.
+ * </p>
+ * <p>
+ * The init parameter 'quartz:start-delay-seconds' can be used to specify the amount of time to wait after initializing the scheduler before
+ * scheduler.start() is called.
+ * </p>
+ *
+ * @author James House
+ * @author Chuck Cavaness
+ * @author John Petrocik
+ * @author timmolter
+ */
+public class SundialInitializerListener implements ServletContextListener {
+
+  private boolean performShutdown = true;
+
+  private boolean waitOnShutdown = false;
+
+  private final Logger logger = LoggerFactory.getLogger(getClass());
+
+  @Override
+  public void contextInitialized(ServletContextEvent servletContextEvent) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 15. By comparison, 99% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test. We recommend that you simplify this method or break it into multiple methods.